docs/proposals/disk-accounting.md

```diff
@@ -0,0 +1,641 @@
+
+#### Writable Layer
+
+Docker creates a writable layer for every container on the host. Depending on the storage driver, the location and the underlying filesystem of this layer will change.
+
+Any files that the container creates or updates (assuming there are no volumes) will be considered as writable layer usage.
+
+The underlying filesystem is whatever the docker storage directory resides on. It is ext4 by default on most distributions.
+
+#### Container’s logs
+
+Docker engine provides a pluggable logging interface. Kubernetes is currently using the default logging mode which is `local file`. In this mode, the docker daemon stores bytes written by containers to their stdout or stderr, to local disk. These log files are contained in a special directory that is managed by the docker daemon. These logs are exposed via `docker logs` interface which is then exposed via kubelet and apiserver APIs. Currently, there is a hard-requirement for persisting these log files on the disk.
+
+#### Local Volumes
+
+Volumes are slightly different from the other use cases local disk. They are pod scoped. Their lifetime is tied to that of a pod. Due to this property accounting of volumes will also be at the pod level.
+
+As of now, the volume types that use local disk directly are ‘HostPath’, ‘EmptyDir’, and ‘GitRepo’. Everything else is a network based volume.
+
+‘HostPath’ volumes map in existing directories in the host filesystem into a pod. Kubernetes manages only the mapping. It does not manage the source on the host filesystem.
+
+In addition to this, the changes introduced by a pod on the source of a hostPath volume is not cleaned by kubernetes once the pod exits. Due to these limitations, we will have to account hostPath volumes to system overhead. We should explicitly discourage use of HostPath in read-write mode.
+
+`EmptyDir` and `GitRepo` volumes map to a directory on the host root filesystem, that is managed by Kubernetes (kubelet). Their contents are erased as soon as the pod exits. Tracking and potentially restricting usage for volumes is possible.
+
+### Docker storage model
+
+Before we start exploring solutions, let’s get familiar with how docker handles storage for images, writable layer and logs.
+
+On all storage drivers, logs are stored under `<docker root dir>/containers/<container-id>/`
+
+The default location of the docker root directory is `/var/lib/docker`.
+
+Volumes are handled by kubernetes.
+
+Container images and writable layers are managed by docker and their location will change depending on the storage driver. Each image layer and writable layer is referred to by an ID. The image layers are read-only. Once saved, existing writable layers can be frozen. Saving feature is not of importance to kubernetes since it works only on immutable images.
+
+*Note: Image layer IDs can be obtained by running `docker history -q --no-trunc <imagename>`*
+
+##### Aufs
```

Comment:
This section on aufs, devicemapper, and overlayfs is awesome to have in this doc, :100: